docs/en/advanced_config/bootloader_secure_boot.md

```diff
@@ -0,0 +1,323 @@
+
+## Enabling Secure Boot
+
+### Prerequisites
+
+Before enabling secure boot, you need:
+
+1. A board configuration that supports secure boot (NuttX-based flight controllers)
+2. Understanding of the build system and firmware signing process
+3. Secure storage for your private keys
+4. A workflow for signing firmware images
+
+### Board Configuration
+
+To enable secure boot on your board:
+
+1. **Enable security in bootloader**: Add to your board's bootloader configuration file (e.g., `boards/[vendor]/[board]/nuttx-config/bootloader/defconfig`):
+
+   ```sh
+   CONFIG_BOOTLOADER_USE_SECURITY=y
+   ```
+
+2. **Set crypto algorithm**: Define the signing algorithm in your board's `hw_config.h`:
+
+   ```c
+   #define BOOTLOADER_SIGNING_ALGORITHM CRYPTO_ED25519
+   ```
+
+3. **Configure key storage**: Implement or configure the keystore backend for your board
+4. **Build bootloader with crypto support**: Ensure `PX4_CRYPTO` is defined in your board's CMake configuration
+
+### Key Management
+
+::: warning CRITICAL
+Keep your private keys secure! If private keys are lost, you cannot sign new firmware. If private keys are compromised, unauthorized firmware can be created.
+:::
+
+Key management involves:
+
+1. **Key generation**: Generate cryptographic key pairs (public and private keys)
+2. **Key storage**:
+   - Private keys: Store securely on your build/signing server (never on the device)
+   - Public keys: Store in the device's keystore (accessible to bootloader)
+3. **Key indices**: Each key has an index (0-15) used to reference it in the TOC
+
+#### Generating ED25519 Keys
+
+You can use various tools to generate ED25519 keys.
+Example using Python with the `cryptography` library:
+
+```python
+from cryptography.hazmat.primitives.asymmetric.ed25519 import Ed25519PrivateKey
+from cryptography.hazmat.primitives import serialization
+
+# Generate private key
+private_key = Ed25519PrivateKey.generate()
+
+# Get public key
+public_key = private_key.public_key()
+
+# Serialize private key
+private_bytes = private_key.private_bytes(
+    encoding=serialization.Encoding.Raw,
+    format=serialization.PrivateFormat.Raw,
+    encryption_algorithm=serialization.NoEncryption()
+)
+
+# Serialize public key
+public_bytes = public_key.public_bytes(
+    encoding=serialization.Encoding.Raw,
+    format=serialization.PublicFormat.Raw
+)
+
+# Save keys to files
+with open('private_key.bin', 'wb') as f:
+    f.write(private_bytes)
+
+with open('public_key.bin', 'wb') as f:
+    f.write(public_bytes)
+```
+
+#### Installing Public Keys on Device
+
+Public keys must be installed in the device's keystore. The exact method depends on your board's keystore implementation:
+
+- Some boards use One-Time Programmable (OTP) memory
+- Some boards use secure flash storage
+- Some implementations use a stub keystore for development/testing
+
+Refer to your board's keystore implementation in `src/drivers/stub_keystore/` or board-specific keystore drivers.
```

Comment:
I think this is probably accurate but unhelpful/useless.

Keys can be copied into the firmware using the CONFIG_PUBLIC_KEYn etc. But that MIGHT only be for the software keystore, which is why we need to document the options for our keystore 
- can you help with that? 
- What are our keystore options?
- How do you get keys into each type of supported keystore? The answer might be "use tools that are supplied by your OEM" in some cases, but we should say that.


Is there